quartz/components/scripts/darkmode.inline.ts

```diff
@@ -2,15 +2,19 @@ const userPref = window.matchMedia("(prefers-color-scheme: light)").matches ? "l
 const currentTheme = localStorage.getItem("theme") ?? userPref
 document.documentElement.setAttribute("saved-theme", currentTheme)
 
+const emitThemeChangeEvent = (theme: "light" | "dark") => {
+  const event = new CustomEvent("themechange", {
+    detail: { url: document.body.dataset.slug, theme: theme },
```

Comment:
```suggestion
    detail: { theme },
```

don't need `url` here

Reply:
we should strongly type this,

add an event to `CustomEventMap` in `index.d.ts` and make `event` -> `event: CustomEventMap["themechange"]`

Reply:
Done!